### PR TITLE
Change 2-of-2 integration test to 2-of-3

### PR DESCRIPTION
**Pull request by rubensayshi**
add failing unittest to prove the error I'm trying to fix in https://github.com/bitcoinjs/bitcoinjs-lib/pull/369

### CODE REVIEW

File: test/integration/multisig.js [hunk 1]
```diff
@@ -77,4 +77,63 @@ describe('bitcoinjs-lib (multisig)', function () {
...
+        var txb = new bitcoin.TransactionBuilder()
+        txb.addInput(unspent.txId, unspent.vout)
+        txb.addOutput(targetAddress, 1e4)
+
+        // sign with 1st and 3rd key
+        txb.sign(0, privKeys[0], redeemScript)
+        txb.sign(0, privKeys[2], redeemScript)
+
+        // broadcast our transaction
+        blockchain.transactions.propagate(txb.build().toHex(), function (err) {
```
Comment (dcousens) [inline test/integration/multisig.js]: This should fail.

**Comment by dcousens** [comment]
I can't add this test as is, because technically the test is taking advantage of code that should be failing elsewhere.

We do however need to add a test to ensure that `txb.build().toHex()` fails when a signature is missing.  Infact, I'm not sure why it isn't already.

**Comment by dcousens** [comment]
Disregard previous comments, accidentally read this as `3 of 3`, not `2 of 3`.

**Comment by dcousens** [comment]
@rubensayshi I have added https://github.com/bitcoinjs/bitcoinjs-lib/commit/c79fecffa3ded9ee26db8c777b72a58d09ed8a48 to show case this test without the need for it to be an integration test.

What is shown there is the expected `txHex`,  it verifies that no OP_0 should be present in the scriptSig; and as such, before https://github.com/bitcoinjs/bitcoinjs-lib/commit/707376c81eefea312392c0dc9f6868e9aff3eb3d was a failing test. 

Please ACK this and I'll close this PR. :)

**Comment by rubensayshi** [comment]
hmm, I changed the 2of2 integration test to 2of3, I think that's a better test case since it has a greater chance of failing (for various reasons)

**Comment by dcousens** [comment]
@rubensayshi I'm OK with this change, but I don't think it covers any less or more than `2-of-2`, it simply covers the other case (fully signed vs partially signed). 

However this does show case a better example of multisig to the typical user, so I'll update the README example after to merge.

**Comment by dcousens** [comment]
@rubensayshi to follow that line of thought, [I changed it](https://github.com/bitcoinjs/bitcoinjs-lib/commit/29b93b8a5c393a21865a359d2491faec5664d2e8) to a less traditional `2-of-4` which would better illustrate some parts of the API and provide a larger contrast to the example above it, the `2-of-2`.

**Comment by rubensayshi** [comment]
awesome!

**Pull request merged**